src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -620,4 +620,55 @@ BOOST_AUTO_TEST_CASE(CustomLayerTest)
   arma::mat prediction = arma::zeros<arma::mat>(1, predictionTemp.n_cols);
 }
 
+/**
+ * Test the overload of Forward function which allows partial forward pass.
+ */
+BOOST_AUTO_TEST_CASE(PartialForwardTest)
+{
+  FFN<NegativeLogLikelihood<>, RandomInitialization>* model =
+      new FFN<NegativeLogLikelihood<>, RandomInitialization>();
```

Comment:
Ok, now I understand what's happened here.  Changing `model` to be a manually-allocated pointer to an `FFN` isn't necessary, my concern originally was that I saw that `addModule` and `linearNoBiasModule` were being allocated without any `delete` later.  So I thought that the memory that was being allocated would never be freed.

However after digging into it, the particular way this code currently works (see #1489), `model` will actually delete `addModule` and `linearNoBiasModule` when its destructor is called.  That destructor gets called regardless of whether `model` is allocated with `new` or not---when using `new`, the destructor is called when we call `delete`.  But if we use a local variable for `model`, then the destructor is called when it goes out of scope (at the end of the function) and the memory is freed there too.

So you can revert to a local `FFN` for `model` if you like, but this code works either way and doesn't leak right now.  Depending on what we do for #1489, this may need to be adjusted later, but I don't think we need to worry about it.

tl;dr: I was incorrect about the memory leak, sorry about that. :)

Reply:
Okay, I guess it's okay for me to revert to the second last commit then. Yeah, even I forgot that the FFN class deletes everything.